src/test.jl

```diff
@@ -610,10 +610,10 @@ module TestPaths
         @testset "mkdir" begin
             garply = ps.root / "corge" / "grault" / "garply"
             @test_throws ErrorException mkdir(garply)
-            mkdir(garply; recursive=true)
+            @test mkdir(garply; recursive=true) == garply
```

Comment:
Irrelevant to this PR.

But why is this test file in `src/`?

Reply:
I believe it's here so other path types in downstream packages can test their compliance with the interface

Reply:
Oh cool, good to know!

Reply:
Yep, we use it AWSS3